woocommerce/woocommerce
# [api tests] Isolate test context for fix flakiness in product crud api tests

### Changes proposed in this Pull Request:

More context: p1725034149908379-slack-C01BWDDTGKX

All tests in product-crud.test.js were sharing a single global productId which was updated and eventually deleted by a tests creating all sorts of issues.

Updated the spec to have each test or test describe block create their own test product to isolate the test context and avoid flakiness.

### How to test the changes in this Pull Request:

1. CI green.
2. Locally, run the affected test (will run a single worker):
```
pnpm test:api products-crud.test.js
```

3. Locally, run all tests (should run with 4 workers):
```
pnpm test:api
```
